DAH-1175 pricing table

kramduckner opened the pull request:
[DAH-1175](https://sfgovdt.jira.com/browse/DAH-1175)

[DAH-1175]: https://sfgovdt.jira.com/browse/DAH-1175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kramduckner commented on app/javascript/modules/listings/BuyHeader.scss:
```diff
@@ -28,4 +28,4 @@
   @apply md:text-4xl;
-}
\ No newline at end of file
+}
```
unsure what this change is about - [I'm seeing ](https://github.com/SFDigitalServices/sf-dahlia-web/blob/main/app/javascript/modules/listings/BuyHeader.scss#L31)that the main branch has that closing bracket.

chadbrokaw-exygy reviewed (approved):
Looks good to me! Just a few comments, but no blockers.

Comparing to the designs (and sorry if this has been discussed and I missed it) there are two differences that I noticed:

On the sales table, there is another border beneath the table header. These aren't in the designs.
<img width="653" alt="Sale unit has a border beneath the header info" src="https://user-images.githubusercontent.com/122493877/223566906-df142446-8190-4bad-97e8-251a3adf292f.png">

Also in the designs, the bottom of the accordion should have a border radius?

kramduckner commented on app/javascript/__tests__/modules/listingDetails/ListingDetailsPricingTable.test.tsx:
```diff
@@ -1,19 +1,51 @@
+
+    const tree = renderer.create(<ListingDetailsPricingTable listing={closedRentalListing} />)
+
+    await act(() => new Promise((resolve) => setTimeout(resolve)))
```
Yup! It's kind of odd. I added a comment.

kramduckner commented on app/javascript/__tests__/modules/listingDetails/ListingDetailsPricingTable.test.tsx:
```diff
@@ -1,19 +1,51 @@
 import { ListingDetailsPricingTable } from "../../../modules/listingDetails/ListingDetailsPricingTable"
 import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
-import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"
+// import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"
```
I'm always a fan of getting rid of commented code! But this is a little unique as we ended up rolling back some of the AC of the ticket, so extra work was done than what was needed. I added future tickets for the habitat listings, so since we'll work a habitatListing test soon, I vote to keep it in.

kramduckner commented on app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx:
```diff
@@ -1,66 +1,227 @@
+      cellSubText: `${unitSummary?.availability} ${t("t.available")}`,
+    },
+    income: {
+      cellText: `$${unitSummary.absoluteMinIncome?.toLocaleString()} to $${unitSummary.absoluteMaxIncome?.toLocaleString()}`,
```
I added a couple of those to the translation doc! I talked with Christine about the others, and we were of a mind that they didn't need interpolated translations, but the more I think about it, I say we should just add all of them. Iirc, it won't cost extra. I'll do it! Thanks for bring that up.

kramduckner commented on app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx:
```diff
@@ -1,66 +1,227 @@
+            }
+
+        return {
+          header: t("listings.stats.upToPercentAmi", {
```
Good catch! I'm going to add a ticket for this. I think it has to do with some new legislation and I'm unsure which fields to check.

kramduckner commented:
@chadbrokaw-exygy Thanks a million for reviewing! I fixed one of the styling issues. I can't remember exactly, but I think Yindi said it was ok that we have that extra border. I hope you feel I addressed all the other comments. If not, happy to make more tickets.
